firewood/src/storage/linear/proposed.rs

```diff
@@ -32,16 +33,40 @@ impl<P: ReadLinearStore, M> Proposed<P, M> {
     }
 }
 
+#[derive(Debug)]
+struct Layer<'a, P: ReadLinearStore> {
+    parent: Arc<LinearStore<P>>,
+    diffs: &'a BTreeMap<u64, Box<[u8]>>,
```

Comment:
not sure how this will work with `Proposed`, which needs both the old and the new data. The old sticks around in case it commits, and gets migrated into a `Current`. If we don't keep the old data, we'd have have to recreate it on commit.